Enhance acronym ESLint rule to enforce lowercase when applicable

johnPhillips opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses. -->
Addresses issue #2195

## Relevant technical choices

<!-- Please describe your changes. -->

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

tofumatt commented:
@googlebot I consent.

tofumatt commented on packages/eslint-plugin/utils.js:
```diff
@@ -100,14 +101,29 @@ function isImported( node ) {
-		[ 'ArrowFunctionExpression', 'FunctionDeclaration', 'FunctionExpression' ].includes( node.type )
-	);
+	if (
+		node?.type === 'Identifier' && (
+			hasFunctionType( node?.parent?.type ) ||
+			hasFunctionType( node?.parent?.init?.type )
+		)
+	) {
+		return true;
+	}
+
+	const isFunctionDeclaration = hasFunctionType( node?.type );
```
I think we weren't using optional chaining in these files before because our version of node was too old, but now that we require Node 14 via NVM this should be fine. 👍🏻

aaemnnosttv reviewed (commented):
LGTM!

@johnPhillips it looks like all the changes were from renaming the rule, is that correct? Were there any actual violations that needed fixing?

Otherwise, the only thing needed here is a file conflict to be resolved.

johnPhillips commented:
@aaemnnosttv the bulk of the changes were just the renaming, yes. I also added a couple of extra checks to the rule as per the AC. There weren't any violations that I can remember, so it was pretty clean.

aaemnnosttv reviewed (approved):
Great!

aaemnnosttv reviewed (changes requested):
Oops! I just tried this out and found that it did not detect a few cases where it should.

![image](https://user-images.githubusercontent.com/1621608/106600479-27de9b80-6563-11eb-8ee2-f1af10e9140b.png)

I pushed an update to the test case for this rule so it should be as simple as getting the tests to pass again from here 😄

aaemnnosttv commented:
@googlebot I consent.

johnPhillips commented:
@aaemnnosttv CC @tofumatt 
I fixed the broken test there, but it seems to have been down to this line:
```js
// If the name of this variable is the same length as the acronym,
// it should be lowercase or uppercase.
- if ( name.length === acronym.length && ( acronymMatch === acronym.toLowerCase() || acronymMatch !== acronym ) ) {
+ if ( name.length === acronym.length && ( acronymMatch === acronym.toLowerCase() || acronymMatch === acronym.toUpperCase() ) ) {
```
I couldn't figure out what the original intent was with the ` || acronymMatch !== acronym` check here. It didn't _seem_ necessary, so I removed it. I hope I didn't break something - more than happy to add it back in if I did 👍

aaemnnosttv reviewed (approved):
Super, thanks!
